src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -574,28 +574,29 @@ private function parseDefinition($id, $service, $file, array $defaults)
     /**
      * Parses a callable.
      *
-     * @param string|array $callable  A callable
-     * @param string       $parameter A parameter (e.g. 'factory' or 'configurator')
-     * @param string       $id        A service identifier
-     * @param string       $file      A parsed file
+     * @param string|array $callable  A callable reference
+     * @param string       $parameter The type of callable (e.g. 'factory' or 'configurator')
      *
      * @throws InvalidArgumentException When errors occur
      *
      * @return string|array|Reference A parsed callable
      */
-    private function parseCallable($callable, $parameter, $id, $file)
+    private function parseCallable($callable, string $parameter, string $id, string $file)
     {
         if (\is_string($callable)) {
             if ('' !== $callable && '@' === $callable[0]) {
                 if (false === strpos($callable, ':')) {
                     return [$this->resolveServices($callable, $file), '__invoke'];
                 }
-                throw new InvalidArgumentException(sprintf('The value of the "%s" option for the "%s" service must be the id of the service without the "@" prefix (replace "%s" with "%s").', $parameter, $id, $callable, substr($callable, 1)));
+
+                throw new InvalidArgumentException(sprintf('The value of the "%s" option for the "%s" service must be the id of the service without the "@" prefix (replace "%s" with "%s" in "%s").', $parameter, $id, $callable, substr($callable, 1), $file));
             }
 
             if (false !== strpos($callable, ':') && false === strpos($callable, '::')) {
                 $parts = explode(':', $callable);
 
+                @trigger_error(sprintf('Using short %s syntax for service "%s" is deprecated since Symfony 4.4, use "[\'@%s\', \'%s\']" instead.', $parameter, $id, ...$parts), E_USER_DEPRECATED);
```

Comment:
If `$parameter` is something that the user types ... then we should replace `short %s syntax` by `short "%s" syntax` to avoid potential confusions.

Reply:
it can only be factory of configurator, so no issue here.